packages/rich-text/src/create.js

```diff
@@ -268,16 +264,6 @@ function filterRange( node, range, filter ) {
 	return { startContainer, startOffset, endContainer, endOffset };
 }
 
-/**
- * Collapse any whitespace used for HTML formatting to one space character,
- * because it will also be displayed as such by the browser.
- *
- * @param {string} string
- */
-function collapseWhiteSpace( string ) {
-	return string.replace( /[\n\r\t]+/g, ' ' );
```

Comment:
This is sort of replaced by logic in the paste handler

https://github.com/WordPress/gutenberg/blob/acd54770808254a650e296f88d25a7e0da5e8596/packages/blocks/src/api/raw-handling/html-formatting-remover.js#L50